File: YourHome.API/ClientApp/src/app/offer/models/location.ts
export interface Location {
    city: string;
    district: string;
    voivodeship: string;
    houseNumber: string;
    apartmentNumber: string;
    coordinates : Coordinates | null;
}

export class NewLocation{
    city: string;
    district: string;
    voivodeship: string;
    houseNumber: string;
    apartmentNumber: string;
}

export const EMPTY_LOCATION = {
    city: "",
    district: "",
    voivodeship: "",
    houseNumber: "",
    apartmentNumber: "",
    coordinates : null
}

export interface Coordinates{
    latitude : number,
    longitude : number
}

File: YourHome.Core/Models/Domain/Location.cs
namespace YourHome.Core.Models.Domain
{
    public class Location
    {
        public string City { get; set; }
        public string District { get; set; }
        public string Voivodeship { get; set; }
        public string HouseNumber { get; set; }
        public string ApartmentNumber { get; set; }
        public string Street { get; set; }
        public Coordinates Coordinates { get; set; }
    }
}
